src/prober.rs

```diff
@@ -0,0 +1,488 @@
+// Copyright 2017 Red Hat, Inc.
+
+// Licensed under the MIT license <LICENSE or
+// http://opensource.org/licenses/MIT> This file may not be copied, modified,
+// or distributed except according to those terms.
+
+use crate::{
+    error::{c_result, BlkIdError, BlkIdResult},
+    part_list::PartList,
+    path_to_cstring,
+    topology::Topology,
+    PartitionsFlags, SuperblocksFlags,
+};
+use blkid_sys::*;
+use std::{
+    collections::HashMap,
+    ffi::{CStr, CString},
+    path::Path,
+    ptr,
+};
+
+/// Low-level probing setting
+///
+/// The probing routines are grouped together into separate chains. Currently, the library provides
+/// superblocks, partitions and topology chains.
+///
+/// The probing routines is possible to filter (enable/disable) by type (e.g. fstype "vfat" or
+/// partype "gpt") or by usage flags (e.g. BLKID_USAGE_RAID). These filters are per-chain. Note that
+/// always when you touch the chain filter the current probing position is reset and probing starts
+/// from scratch. It means that the chain filter should not be modified during probing, for example
+/// in loop where you call [`Self::do_probe`].
+///
+/// The probing routines inside the chain are mutually exclusive by default - only few probing
+/// routines are marked as "tolerant". The "tolerant" probing routines are used for filesystem
+/// which can share the same device with any other filesystem. The [`Self::safeprobe`] checks for
+/// the "tolerant" flag.
+///
+/// The `superblocks` chain is enabled by default. The all others chains is necessary to enable by
+/// `enable_'CHAINNAME'()`.
+pub struct Prober(pub(crate) blkid_probe);
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum ProbeState {
+    Success,
+    Done,
+    NothingDetected,
+    Ambivalent,
```

Comment:
I'd probably name this `Unknown` but this is ok also

Reply:
Unfortunately, I don't understand meaning of `Ambivalent` at all. I just used that because in blkid doc the same. If you can explain me what does it mean I will be glad.
I think you better understands meaning of this word and we can use `Unknown` instead.

Reply:
Ahh I see. I didn't realize Blkid was using Ambivalent also. If that's the case I would just leave it as is